.cascade-code/left-nav-refactor/_components/left_nav/left_nav_format

```diff
@@ -0,0 +1,125 @@
+        #set ( $siblingTitle = $_EscapeTool.xml($sibling.getChild("display-name").value) )
+        #set ( $siblingType = $sibling.getName() )
+        #set ( $hideSibling = $_XPathTool.selectNodes($sibling, "dynamic-metadata[1][value='Yes'] or system-page[name = 'index']/dynamic-metadata[1][value='Yes']") )
+        #set ( $children = $_XPathTool.selectNodes($sibling, "system-page[name != 'index'] | system-folder") )
+        #set ( $hiddenChildren = $_XPathTool.selectNodes($sibling, "system-page[name != 'index']/dynamic-metadata[1][value='Yes'] | system-folder[dynamic-metadata[1][value='Yes']]") )
+        
+        #if ( $children.size() > $hiddenChildren.size() )
+            #set ( $siblingHasChildren = true )
+        #else
+            #set ( $siblingHasChildren = false )
+        #end
+         
+        #if ( $siblingType == "system-folder" && $hideSibling == [false] )
+        
+            #if ( $current == true )
+                <li class="active">
+            #else
+                <li>
+            #end
+
+            <a href="${siblingPath}/index" class="s-f-idx">
+                $siblingTitle
+                #if ( $siblingHasChildren && $current == false )
+                    <span class="plus">+</span>
+                #end
+            </a>
+            #if ( $siblingHasChildren )
+                #displayChildren( $children )
+            #end
+            </li>
+            
+        #elseif ( $siblingType == "system-page" && $hideSibling == [false] )
+
+            #if ( $current == true )
+                <li class="active">
+            #else
+                <li>
+            #end
+            
+            <a href="${siblingPath}" title="${siblingTitle}" class="s-p">
+                $siblingTitle 
+            </a>
+            </li>
+
+        #end
+    #end
+    </ul>
+#end
+
+#set ( $parentDisplayName = $parent.getChild("display-name") )
+
+### if fails to find parent page, don't do this LI...gives error (eg parent folder unindexed or not right metadata):
+#if ( $parentDisplayName.value )
+    #set ( $parentTitle = $_EscapeTool.xml( $parentDisplayName.value ) )
+    <div class="leftTitle">
+        <ul>
+            <li class="title">
+            <a href="$parent.getChild("link").value" title="${parentTitle}">
+                $parentTitle
+            </a>
+            </li>
+        </ul>
+    </div>
+#end
+
+#if( $current )       
+    #set ( $siblings = $_XPathTool.selectNodes($currentPage, "//system-folder[system-page[@current and name = 'index']] | //system-page[@current and name != 'index'] | //system-folder[system-page[@current and name = 'index']]/following-sibling::*[name != 'index'] | //system-folder[system-page[@current and name = 'index']]/preceding-sibling::*[name != 'index'] | //system-page[@current and name != 'index']/following-sibling::*[name != 'index'] | //system-page[@current and name != 'index']/preceding-sibling::*[name != 'index']") )
+    #showPage( $siblings )
```

Comment:
What is `showPage` really doing here? Building the left-nav menu? Would `buildLeftNavMenu` be more descriptive?

Reply:
I was thinking of just changing the `showPage` macro to `displaySiblings`, because it goes through all the siblings and if the sibling has children, calls `displayChildren` macro.